### PR TITLE
Store Playlist Info

### PR DESCRIPTION
**Pull request by TheRealPikachu**
**Please describe the changes this PR makes and why it should be merged:**

**Status**

- [ ] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes

**Semantic versioning classification:**

- [ ] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: actions/store_playlist_info_MOD.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
+module.exports = {
+
```
Comment (almeidx) [inline actions/store_playlist_info_MOD.js:2]:
```suggestion
```

File: actions/store_playlist_info_MOD.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  subtitle (data) {
+    return 'Store YouTube playlist information.'
+  },
+
+  variableStorage (data, varType) {
+    const type = parseInt(data.storage)
+    if (type !== varType) return
+    const dataType = 'Playlist Info'
+    return ([data.varName, dataType])
```
Comment (almeidx) [inline actions/store_playlist_info_MOD.js:14]:
```suggestion
    if (parseInt(data.storage) !== varType) return
    return ([data.varName, 'Playlist Info'])
```

File: actions/store_playlist_info_MOD.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    const namelist = JSON.stringify(names)
+
+    const datas = await ytlist(url, ['id', 'name', 'url'])
+    const datalist = JSON.stringify(datas)
+
+    const durationis = await ytlist(url, 'duration')
+    const durationlist = JSON.stringify(durationis)
+
+    const ids = await ytlist(url, 'id')
+    const idlist = JSON.stringify(ids)
```
Comment (almeidx) [inline actions/store_playlist_info_MOD.js:74]: You only need to fetch all of these when their options are selected, so put them inside each case on the switch and adapt the code accordingly

File: actions/store_playlist_info_MOD.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      case 4:
+        result = idlist
+        break
+    }
+
+    this.storeValue(result, storage, varName, cache)
+    this.callNextAction(cache)
+  },
+
+  mod (DBM) {}
```
Comment (almeidx) [inline actions/store_playlist_info_MOD.js:98]:
```suggestion
  mod () {}
```

**Comment by TheRealPikachu** [comment]
> Now that I'm actually looking at this, why exactly is everything being JSON.stringify'ed?

just in case, I'm gonna remove that

**Comment by TheRealPikachu** [comment]
> ... Obviously it's going to send [object Object] in the chat, because you're not supposed to just send the object in the chat..

right

File: actions/store_playlist_info_MOD.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+  async action (cache) {
+    const data = cache.actions[cache.index]
+    const storage = parseInt(data.storage)
+    const varName = this.evalMessage(data.varName, cache)
+    const query = this.evalMessage(data.query, cache)
+    const INFO = parseInt(data.info)
+    const ytlist = require('youtube-playlist')
+    const url = query
+    let result = 5
```
Comment (almeidx) [inline actions/store_playlist_info_MOD.js:56]:
```suggestion
    let result
```

File: actions/store_playlist_info_MOD.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+    switch (INFO) {
+      case 0:
+        const datalist = await ytlist(url, ['id', 'name', 'url'])
+        result = datalist
```
Comment (almeidx) [inline actions/store_playlist_info_MOD.js:61]:
```suggestion
        result = await ytlist(url, ['id', 'name', 'url'])
```

File: actions/store_playlist_info_MOD.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        break
+      case 1:
+        const urllist = await ytlist(url, 'url')
+        result = urllist
```
Comment (almeidx) [inline actions/store_playlist_info_MOD.js:65]:
```suggestion
        result = await ytlist(url, 'url')
```

File: actions/store_playlist_info_MOD.js [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        break
+      case 2:
+        const namelist = await ytlist(url, 'name')
+        result = namelist
```
Comment (almeidx) [inline actions/store_playlist_info_MOD.js:69]:
```suggestion
        result = await ytlist(url, 'name')
```

File: actions/store_playlist_info_MOD.js [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+        break
+      case 3:
+        const durationlist = await ytlist(url, 'duration')
+        result = durationlist
```
Comment (almeidx) [inline actions/store_playlist_info_MOD.js:73]:
```suggestion
        result = await ytlist(url, 'duration')
```

File: actions/store_playlist_info_MOD.js [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+        break
+      case 4:
+        const idlist = await ytlist(url, 'id')
+        result = idlist
```
Comment (almeidx) [inline actions/store_playlist_info_MOD.js:77]:
```suggestion
        result = await ytlist(url, 'id')
```

File: actions/store_playlist_info_MOD.js [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+  init () {},
+
+  async action (cache) {
+    const data = cache.actions[cache.index]
+    const storage = parseInt(data.storage)
+    const varName = this.evalMessage(data.varName, cache)
+    const query = this.evalMessage(data.query, cache)
+    const INFO = parseInt(data.info)
+    const ytlist = require('youtube-playlist')
+    const url = query
```
Comment (almeidx) [inline actions/store_playlist_info_MOD.js:55]: ?

Comment (TheRealPikachu) [reply actions/store_playlist_info_MOD.js:55]: fixing

File: actions/store_playlist_info_MOD.js [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+
+  init () {},
+
+  async action (cache) {
+    const data = cache.actions[cache.index]
+    const storage = parseInt(data.storage)
+    const varName = this.evalMessage(data.varName, cache)
+    const url = this.evalMessage(data.query, cache)
+    const INFO = parseInt(data.info)
+    const ytlist = require('youtube-playlist')
```
Comment (almeidx) [inline actions/store_playlist_info_MOD.js:54]: Use Mods.require()

File: docs/mods.json [hunk 13]
```diff
@@ -959,6 +959,11 @@
     "name": "Store YouTube Channel Info",
     "section": "YouTube Tools"
   },
+  "store_playlist_info_MOD.js": {
```
Comment (almeidx) [inline docs/mods.json:962]: Put this in the right place. (Alphabetically sorted)

**Pull request merged**